### ISSUE TITLE
Re-exported type reported as unused with `ignoreExportsUsedInFile`

### ISSUE THREAD

**Issue by haines** [webpro/knip#622]
Repro: https://stackblitz.com/edit/github-hqqlbk?file=component%2Findex.ts

We have a standard module template for our React components, so that every component (at least) exports its props and the component itself, like

```ts
// component.ts
export interface ComponentProps { ... }
export default function Component(props: ComponentProps) { ... }
```

This works fine with Knip even if nothing imports the props, as long as `ignoreExportsUsedInFile` is set to `true`.

However, it doesn't work on our wrapper components, where we re-export the props from an internal file

```ts
// index.ts
import { ComponentProps } from "./component";

export type { ComponentProps };
export default function Wrapper(props: ComponentProps) { ... }
```

I would have expected in this case that `ComponentProps` would not be reported as an unused export from `index.ts` because of `ignoreExportsUsedInFile`, but it was:

```console
$ npm run knip

> @knip/templates-basic@1.0.0 knip
> knip

Unused exports (1)
ComponentProps  unknown  component/index.ts:4:9
```

**Comment by webpro** [comment]
Thanks for the report, @haines. The issue is that the _usage_ of `ComponentProps` refers to the import, not the export. But a strict re-export wouldn't be an import, and therefore no usage would be possible. So I've tried to come up with a fix.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.12.2. See [Release 5.12.2](https://github.com/webpro/knip/releases/tag/5.12.2) for release notes._

_Using Knip in a commercial project? Please consider [sponsoring me](https://github.com/sponsors/webpro)._

**Comment by haines** [comment]
Awesome, thanks for fixing it so quickly! I've tried out 5.12.2 and it no longer reports these cases 🎉
